public/app/features/library-panels/components/PanelLibraryOptionsGroup/PanelLibraryOptionsGroup.tsx

```diff
@@ -11,29 +12,47 @@ import { LibraryPanelsView } from '../LibraryPanelsView/LibraryPanelsView';
 import { PanelQueriesChangedEvent } from 'app/types/events';
 import { LibraryPanelDTO } from '../../types';
 import { toPanelModelLibraryPanel } from '../../utils';
+import { connect, MapDispatchToProps } from 'react-redux';
+import { changePanelPlugin } from 'app/features/dashboard/state/actions';
 
 interface Props {
   panel: PanelModel;
   dashboard: DashboardModel;
+  onChange: () => void;
+}
+interface DispatchProps {
+  changePanelPlugin: typeof changePanelPlugin;
 }
 
-export const PanelLibraryOptionsGroup: React.FC<Props> = ({ panel, dashboard }) => {
+export const PanelLibraryOptionsGroupUnconnected: React.FC<Props & DispatchProps> = ({
+  panel,
+  dashboard,
+  onChange,
+  changePanelPlugin,
+}) => {
   const styles = useStyles(getStyles);
   const [showingAddPanelModal, setShowingAddPanelModal] = useState(false);
 
   const useLibraryPanel = (panelInfo: LibraryPanelDTO) => {
+    const panelTypeChanged = panel.type !== panelInfo.model.type;
     panel.restoreModel({
-      ...panelInfo.model,
+      ...omit(panelInfo.model, 'type'),
       ...pick(panel, 'gridPos', 'id'),
       libraryPanel: toPanelModelLibraryPanel(panelInfo),
     });
```

Comment:
I just realized that this code looks very similar to some code I moved here: https://github.com/grafana/grafana/blob/5cbc58fd97e0ebcbaf41cbb57b8f6714a7a340fd/public/app/features/library-panels/utils.ts. Would be great if we could merge some of these functions in future PRs. Seems a bit error prone to have almost the same functionality in two places. WDYT?